casper/src/main/resources/PoS.rhox

```diff
@@ -531,7 +547,86 @@ new  PoS,
                   }
                }
              }
+            } |
+
+            // Initialize pending rewards state for each validator
+            contract PendingRewards(@"init", @validatorKeys, @return) = {
+              new createStateKeyCh in {
+                @ListOps!("map", initialActive, *createStateKeyCh, return) |
+                contract createStateKeyCh(@key, @ackCh) = {
+                  TreeHashMap!("set", pendingRewardsMap, key, {}, ackCh)
+                }
+              }
+            } |
+
+            // Sum new pending rewards to existing state
+            contract PendingRewards(@"append", @validatorKey, @newRewards, @return) = {
+              new currentRewardsMaybeCh, currentRewardsCh in {
+                // Get existing pending rewards (per validator)
+                TreeHashMap!("get", pendingRewardsMap, validatorKey, *currentRewardsMaybeCh) |
```

Comment:
Because all keys must be initialized maybe we can use `fastUnsafeGet` to get validator state?

Reply:
Yes, if you know the key is in the treeHashMap, then you can use fastUnsafeGet to retrieve it quickly.  But also, if you know the key is in the map, then do you need the currentRewardsMaybe below?

Reply:
No, I'll change it to use `fastUnsafeGet`.